test/unit/org/apache/cassandra/index/sai/metrics/QueryMetricsTest.java

```diff
@@ -108,6 +111,56 @@ public void testMetricRelease() throws Throwable
         assertThatThrownBy(() -> getTableQueryMetrics(keyspace, table, "TotalQueriesCompleted")).hasCauseInstanceOf(InstanceNotFoundException.class);
     }
 
+    @Test
+    public void testIndexQueryWithPartitionKey() throws Throwable
+    {
+        String table = "test_range_key_type_with_index";
+        String index = "test_range_key_type_with_index_index";
+
+        String keyspace = createKeyspace(CREATE_KEYSPACE_TEMPLATE);
+
+        createTable(String.format(CREATE_TABLE_TEMPLATE, keyspace, table));
+        createIndex(String.format(CREATE_INDEX_TEMPLATE, index, keyspace, table, "v1"));
+
+        int rowsWrittenPerSSTable = 10;
+        int numberOfSSTable = 5;
+        int rowsWritten = 0;
+        int i = 0;
+        for (int j = 0; j < numberOfSSTable; j++)
+        {
+            rowsWritten += rowsWrittenPerSSTable;
+            for (; i < rowsWritten; i++)
+            {
+                execute("INSERT INTO " + keyspace + "." + table + " (id1, v1, v2) VALUES (?, ?, '0')", Integer.toString(i), i);
+            }
+            flush(keyspace, table);
+        }
+
+        waitForIndexQueryable(keyspace, table);
+
+        ResultSet rows1 = executeNet("SELECT id1 FROM " + keyspace + "." + table + " WHERE v1 = 3");
+        assertEquals(1, rows1.all().size());
+
+        ResultSet rows2 = executeNet("SELECT id1 FROM " + keyspace + "." + table + " WHERE id1 = '36' and v1 < 40");
+        assertEquals(1, rows2.all().size());
+
+        ResultSet rows3 = executeNet("SELECT id1 FROM " + keyspace + "." + table + " WHERE id1 > '31' and v1 = 43 ALLOW FILTERING");
+        assertEquals(1, rows3.all().size());
+
+        ResultSet rows4 = executeNet("SELECT id1 FROM " + keyspace + "." + table + " WHERE id1 > '21' and id1 < '34' and v1 = 23 ALLOW FILTERING");
+        assertEquals(1, rows4.all().size());
+
+        ResultSet rows5 = executeNet("SELECT id1 FROM " + keyspace + "." + table + " WHERE v1 < 24 and v1 > 11");
+        assertEquals(12, rows5.all().size());
+
+        ResultSet rows6 = executeNet("SELECT id1 FROM " + keyspace + "." + table + " WHERE v1 = 20 and id1 = '20'");
+        assertEquals(1, rows6.all().size());
+
+        ObjectName oName = objectNameNoIndex("SSTableIndexesHit", keyspace, table, PER_QUERY_METRIC_TYPE);
+        CassandraMetricsRegistry.JmxHistogramMBean o = JMX.newMBeanProxy(jmxConnection, oName, CassandraMetricsRegistry.JmxHistogramMBean.class);
+        assertTrue(o.getMean() < 2);
```

Comment:
I don't think the test is working as expected. Those queries contain non-partition restricted requests, some of them got filtered because of term range (aka. sstable min/max term doesn't queried term). That's why the mean value is below 2. If you look at second query alone, the mean is actually 5, all sstables are hit.

```
            queryContext.sstablesHit += queryView.referencedIndexes
                                        .stream()
                                        .map(SSTableIndex::getSSTable).collect(Collectors.toSet()).size();
```
The `sstablesHit` metrics counter is recording the num of referenced sstables instead of sstables actually being hit. Should we increment the counter inside `V1SearchableIndex#search`?

Reply:
> Should we increment the counter inside V1SearchableIndex#search instead?

That sounds correct to me.

Reply:
I think we might need a new counter here because sstableHit was used in other places.

Reply:
but the test is measuring the wrong thing if we are just incrementing sstablesHit even when we don't read from it

Reply:
`QueryViewBuilder#indexInRange` looks like a better place to put BF filtering, as it already filters sstable by min/max key range. If we put BF there, sstables that are rejected by BF won't be referenced, and `sstableHits` metrics will be correct.

Reply:
I have changed the BF from `v1searchableIndex` to `QueryViewBuilder`.